### PR TITLE
Check oauth config during oauth login

### PR DESCRIPTION
**Pull request by OptimusRhine**
* on oauth login:
  * explicitly check that the given `provider_name` is valid given our enum `providers_list`
  * explicitly check that we have the correct configurations (`CONSUMER_KEY` and `CONSUMER_SECRET`) for the given provider.

### CODE REVIEW

**Comment by OptimusRhine** [comment]
More or less a duplicate of https://github.com/uwcirg/true_nth_usa_portal/pull/1106 . Closing.

File: portal/views/auth.py [hunk 1]
```diff
@@ -276,6 +276,15 @@ def picture_url(result):
         from werkzeug.datastructures import ImmutableMultiDict
         request.args = ImmutableMultiDict()
 
+    prv = 'FB' if (provider_name == 'facebook') else provider_name.upper()
+
+    if (not current_app.config.get('{}_CONSUMER_KEY'.format(prv)) or
```
Comment (ivan-c) [inline portal/views/auth.py]: Would this also catch when `*_CONSUMER_KEY` is not defined at all (as opposed to just being an empty string)?

 I think you may need to add a second/default arg to `get()`...

Comment (pbugni) [inline portal/views/auth.py]: the second arg defines a default - which is itself `None` if not provided.  this code looks to be doing what we need, to me.

File: portal/views/auth.py [hunk 2]
```diff
@@ -24,7 +24,7 @@
 from ..date_tools import FHIR_datetime
 from ..extensions import authomatic, oauth
 from ..models.auth import AuthProvider, Client, Token, create_service_token
-from ..models.auth import validate_client_origin
+from ..models.auth import providers_list, validate_client_origin
```
Comment (pbugni) [inline portal/views/auth.py]: Looks like you are no longer using `providers_list`.

File: portal/views/auth.py (hunk 1 above)
Comment (OptimusRhine) [inline portal/views/auth.py]: @ivan-c shouldn't we be catching cases like that as well?

**Pull request merged**